File: src/3289.the-two-sneaky-numbers-of-digitville.3581/solution.ts
function getSneakyNumbers (nums: number[]): number[] {
    const arr = new Array<boolean>(nums.length - 2);
    const result:number[] = [];
    for (const n of nums) {
        if (arr[n]) {
            result.push(n);
        }
        arr[n] = true;
    }
    return result;
}

File: src/3289.the-two-sneaky-numbers-of-digitville.3581/solution.java
class Solution {
    public int[] getSneakyNumbers(int[] nums) {
        boolean[] flags = new boolean[nums.length-2];
        int[] result = new int[2];
        int index = 0;
        for(int n:nums){
            if(!flags[n]){
                flags[n] = true;
                continue;
            }
            result[index++] = n;
        }
        return result;
    }
}